### PR TITLE
Decouple coordgen and maeparser integrations

### PR DESCRIPTION
**Pull request by greglandrum**
At the moment these two are tightly coupled within the RDKit integration. That's no longer necessary now that coordgen has the default templates built in. 

This PR allows them to be built separately of each other.

This is nice because now the coordgen integration no longer has a boost::iostreams dependency, which means that we can use it in the JS wrappers.

### CODE REVIEW

**Comment by greglandrum** [comment]
The Travis failure isn’t real

File: External/CoordGen/CMakeLists.txt [hunk 1]
```diff
@@ -85,19 +98,15 @@ if(RDK_BUILD_COORDGEN_SUPPORT)
...
   endif(COORDGEN_FORCE_BUILD OR (NOT coordgen_FOUND))
 
-  include_directories(${maeparser_INCLUDE_DIRS})
   include_directories(${coordgen_INCLUDE_DIRS})
 
-  install(FILES DESTINATION ${RDKit_ShareDir}/Data
-          COMPONENT data)
-
-  set(RDK_COORDGEN_LIBS MolAlign ${coordgen_LIBRARIES} ${maeparser_LIBRARIES} ${Boost_LIBRARIES}
+  set(RDK_COORDGEN_LIBS MolAlign ${coordgen_LIBRARIES} ${RDK_MAEPARSER_LIBS} ${Boost_LIBRARIES}
```
Comment (d-b-w) [inline External/CoordGen/CMakeLists.txt:109]: I think we should also drop all this templatefiledir stuff here: https://github.com/rdkit/rdkit/blob/master/External/CoordGen/CoordGen.h#L52

But that may make sense as a separate commit.
